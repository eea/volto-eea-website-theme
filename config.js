import navigationSettings from './src/reducers/navigation/navigation';

export default function applyConfig(config) {
  config.addonReducers = {
    ...config.addonReducers,
    navigationSettings,
  };
  return config;
}