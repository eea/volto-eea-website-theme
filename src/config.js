import biseLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/bise.svg';
import energyLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/energy.svg';
import insituLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/insitu.svg';
import freshwaterLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/freshwater.svg';
import fiseLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/fise.svg';
import ccaLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/cca.svg';
import copernicusLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/copernicus.svg';
import industryLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/industry.svg';
import marineLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/marine.svg';
import eionetLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/eionet.svg';
import eeaLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/images/Header/eea-logo.svg';
import eeaWhiteLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/eea-white.svg';
import climateLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/climate-health.svg';

// TODO: to be consolidated with headerLinks
// Footer.jsx config options
export const footerOpts = {
  header: 'Thematic information platforms',
  logosHeader: 'Managed by',
  contactHeader: 'Contact Us',
  actions: [
    {
      title: 'Privacy',
      link: '/privacy',
    },
    {
      link: '/sitemap',
      title: 'Sitemap',
    },
    {
      link: '/login',
      title: 'CMS Login',
    },
  ],
  copyright: [
    {
      link: '/copyright',
      site: 'EEA',
    },
  ],
  sites: [
    {
      link: 'https://water.europa.eu/marine',
      src: marineLogo,
      alt: 'WISE marine',
    },
    {
      link: 'https://water.europa.eu/freshwater',
      src: freshwaterLogo,
      alt: 'WISE freshwater',
    },
    {
      link: 'https://biodiversity.europa.eu/',
      src: biseLogo,
      alt: 'Biodiversity',
    },
    {
      link: 'https://forest.eea.europa.eu/',
      src: fiseLogo,
      alt: 'Forest information system for europe',
    },
    {
      link: 'https://climate-adapt.eea.europa.eu/observatory',
      src: climateLogo,
      alt: 'European Climate and Health Observatory',
    },
    {
      link: 'https://climate-adapt.eea.europa.eu/',
      src: ccaLogo,
      alt: 'Climate adapt',
    },
    {
      link: 'https://industry.eea.europa.eu/',
      src: industryLogo,
      alt: 'European industrial emissions portal',
    },
    {
      link: 'https://climate-energy.eea.europa.eu/',
      src: energyLogo,
      alt: 'Climate and energy in the EU',
    },
    {
      link: 'https://land.copernicus.eu/',
      src: copernicusLogo,
      alt: 'Copernicus land monitoring service',
    },
    {
      link: 'https://insitu.copernicus.eu/',
      src: insituLogo,
      alt: 'Copernicus in situ',
    },
  ],
  managedBy: [
    {
      link: 'https://www.eea.europa.eu/',
      src: eeaWhiteLogo,
      alt: 'EEA Logo',
      className: 'site logo',
      columnSize: {
        mobile: 6,
        tablet: 12,
        computer: 4,
      },
    },
    {
      link: 'https://www.eionet.europa.eu/',
      src: eionetLogo,
      alt: 'EIONET Logo',
      className: 'eionet logo',
      columnSize: {
        mobile: 6,
        tablet: 12,
        computer: 4,
      },
    },
  ],
  social: [
    {
      name: 'twitter',
      icon: 'ri-twitter-fill',
      link: 'https://twitter.com/euenvironment',
    },
    {
      name: 'facebook',
      icon: 'ri-facebook-box-fill',
      link: 'https://www.facebook.com/European.Environment.Agency',
    },
    {
      name: 'linkedin',
      icon: 'ri-linkedin-fill',
      link: 'https://www.linkedin.com/company/european-environment-agency',
    },
    {
      name: 'youtube',
      icon: 'ri-youtube-fill',
      link: 'https://www.youtube.com/user/EEAvideos',
    },
    {
      name: 'instagram',
      icon: 'ri-instagram-fill',
      link: 'https://www.instagram.com/ourplanet_eu',
    },
    {
      name: 'rss',
      icon: 'ri-rss-fill',
      link: '/subscription/news-feeds',
    },
  ],
  contacts: [
    {
      icon: 'comment outline',
      text: 'About us',
      link: '/about',
      children: [
        {
          link: '/faq',
          text: 'FAQs',
        },
        {
          link: '/careers',
          text: 'Careers',
        },
      ],
    },
    {
      icon: 'comment outline',
      text: 'Contact us',
      link: '/contact-us',
    },
    {
      icon: 'envelope outline',
      text: 'Sign up to our newsletter',
      link: '/newsletter',
    },
  ],
  address: 'Kongens Nytorv 6 1050 Copenhagen K (+45) 33 36 71 00',
};

// Header.jsx config options
export const headerOpts = {
  logo: eeaLogo,
  logoWhite: eeaWhiteLogo,
  partnerLinks: {
    title: 'Environmental information systems',
    links: [
      {
        title: 'Biodiversity Information System for Europe',
        href: 'https://biodiversity.europa.eu/',
      },
      {
        title: 'Climate Adaptation Platform',
        href: 'https://climate-adapt.eea.europa.eu/',
      },
      {
        title: 'Copernicus in situ component',
        href: 'https://insitu.copernicus.eu/',
      },
      {
        title: 'European Industrial Emissions Portal',
        href: 'https://industry.eea.europa.eu/',
      },
      {
        title: 'Forest Information System for Europe',
        href: 'https://forest.eea.europa.eu/',
      },
      {
        title: 'Information Platform for Chemical Monitoring',
        href: 'https://ipchem.jrc.ec.europa.eu/RDSIdiscovery/ipchem/index.html',
      },
      {
        title: 'Marine Water Information System for Europe',
        href: 'https://water.europa.eu/marine',
      },
      {
        title: 'Fresh Water Information System for Europe',
        href: 'https://water.europa.eu/freshwater',
      },
    ],
  },
};

export const languages = [
  { name: 'Български', code: 'bg' },
  { name: 'čeština', code: 'cs' },
  { name: 'Hrvatski', code: 'hr' },
  { name: 'dansk', code: 'da' },
  { name: 'Nederlands', code: 'nl' },
  { name: 'ελληνικά', code: 'el' },
  { name: 'English', code: 'en' },
  { name: 'eesti', code: 'et' },
  { name: 'Suomi', code: 'fi' },
  { name: 'Français', code: 'fr' },
  { name: 'Deutsch', code: 'de' },
  { name: 'magyar', code: 'hu' },
  { name: 'Íslenska', code: 'is' },
  { name: 'italiano', code: 'it' },
  { name: 'Latviešu', code: 'lv' },
  { name: 'lietuvių', code: 'lt' },
  { name: 'Malti', code: 'mt' },
  { name: 'Norsk', code: 'no' },
  { name: 'polski', code: 'pl' },
  { name: 'Português', code: 'pt' },
  { name: 'Română', code: 'ro' },
  { name: 'slovenčina', code: 'sk' },
  { name: 'Slovenščina', code: 'sl' },
  { name: 'Español', code: 'es' },
  { name: 'Svenska', code: 'sv' },
  { name: 'Türkçe', code: 'tr' },
];

export const defaultLanguage = 'en';

export const websiteTitle = 'Site';
export const organisationName = 'European Environment Agency';
export const logoTargetUrl = '/';

export const headerSearchBox = [
  {
    isDefault: true,
    path: '/en/advanced-search',
    placeholder: 'Search...',
  },
  {
    path: '/en/datahub',
    placeholder: 'Search Datahub...',
    description:
      'Looking for more information? Try searching the full EEA website content',
    buttonTitle: 'Go to full site search',
  },
];

export const colors = [
  // Primary & shades
  '#007B6C',
  '#005248',
  '#007B6C',
  '#00A390',
  '#00CCB4',
  '#00F5D8',
  '#85FFF1',
  '#C8FFF8',
  // Secondary & shades
  '#004B7F',
  '#003052',
  '#004B7F',
  '#006BB8',
  '#0083E0',
  '#0A99FF',
  '#47B3FF',
  '#A0D7FF',
  // Tertiary & shades
  '#2E3E4C',
  '#3D5265',
  '#4C677F',
  '#6989A5',
  '#8EA6C2',
  '#ACCAE5',
  '#DAE8F4',
  // Black & White shades
  '#000000',
  '#323232',
  '#67696b',
  '#BCBEC0',
  '#E6E7E8',
  '#F9F9F9',
  '#FFFFFF',
];
