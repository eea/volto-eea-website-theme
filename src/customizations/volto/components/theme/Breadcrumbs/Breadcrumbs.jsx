/**
 * Breadcrumbs components.
 * @module components/theme/Breadcrumbs/Breadcrumbs
 */

import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { useLocation } from 'react-router';
import { getBaseUrl, hasApiExpander } from '@plone/volto/helpers';
import { getBreadcrumbs } from '@plone/volto/actions';
import config from '@plone/volto/registry';

import EEABreadcrumbs from '@eeacms/volto-eea-design-system/ui/Breadcrumbs/Breadcrumbs';

const isContentRoute = (pathname) => {
  const { settings } = config;
  const normalized_nonContentRoutes = settings.nonContentRoutes.map((item) => {
    if (item.test) {
      return item;
    } else {
      return new RegExp(item + '$');
    }
  });

  const isNonContentRoute =
    normalized_nonContentRoutes.findIndex((item) => item.test(pathname)) > -1;

  return !isNonContentRoute;
};

const Breadcrumbs = (props) => {
  const dispatch = useDispatch();
  const { items = [], root = '/' } = useSelector((state) => state?.breadcrumbs);
  // const pathname = useSelector((state) => state.location.pathname);
  const location = useLocation();
  const { pathname } = location;

  const sections = items.map((item) => ({
    title: item.title,
    href: item.url,
    key: item.title,
  }));

  useEffect(() => {
    if (
      !hasApiExpander('breadcrumbs', getBaseUrl(pathname)) &&
      isContentRoute(pathname)
    ) {
      dispatch(getBreadcrumbs(getBaseUrl(pathname)));
    }
  }, [dispatch, pathname]);

  return (
    <React.Fragment>
      <div id="page-header" />
      <EEABreadcrumbs pathname={pathname} sections={sections} root={root} />
    </React.Fragment>
  );
};

export default Breadcrumbs;
