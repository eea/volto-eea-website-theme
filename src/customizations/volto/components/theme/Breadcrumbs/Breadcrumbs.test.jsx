import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';
import config from '@plone/volto/registry';

import Breadcrumbs from './Breadcrumbs';

const mockStore = configureStore();

describe('Breadcrumbs', () => {
  it('renders a breadcrumbs component', () => {
    const store = mockStore({
      breadcrumbs: {
        items: [
          { title: 'Blog', url: '/blog' },
          { title: 'My first blog', url: '/blog/my-first-blog' },
        ],
      },
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Breadcrumbs pathname="/blog" />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders breadcrumbs with contentTypesAsBreadcrumbSection', () => {
    // Mock the config settings
    config.settings = {
      ...config.settings,
      contentTypesAsBreadcrumbSection: ['Folder', 'News Item'],
    };

    const store = mockStore({
      breadcrumbs: {
        items: [
          { title: 'Home', url: '/' },
          { title: 'News', url: '/news', portal_type: 'Folder' },
          {
            title: 'Latest Update',
            url: '/news/latest-update',
            portal_type: 'News Item',
          },
        ],
      },
      intl: {
        locale: 'en',
        messages: {},
      },
      userSession: {
        token: null,
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Breadcrumbs pathname="/news/latest-update" />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
