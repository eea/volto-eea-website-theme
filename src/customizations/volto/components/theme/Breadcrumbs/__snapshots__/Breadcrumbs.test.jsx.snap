// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders a breadcrumbs component 1`] = `
Array [
  <div
    id="page-header"
  />,
  <div
    className="ui vertical attached segment breadcrumbs"
  >
    <div
      className="ui container"
    >
      <nav
        aria-label="Breadcrumbs"
        className="ui tiny breadcrumb"
      >
        <ol
          aria-label="Breadcrumbs navigation"
        >
          <li
            className="breadcrumbs-home-item"
          >
            <a
              className="section"
              href="/"
              onClick={[Function]}
              title="Home"
            >
              <i
                aria-hidden="true"
                className="icon ri-home-2-line breadcrumbs-home-icon"
                onClick={[Function]}
              />
            </a>
          </li>
          <li>
            <div
              className="divider"
            >
              <i
                aria-hidden="true"
                className="icon ri-arrow-right-s-line"
                onClick={[Function]}
              />
            </div>
            <a
              className="section"
              href="/blog"
              onClick={[Function]}
            >
              Blog
            </a>
          </li>
          <li>
            <div
              className="divider"
            >
              <i
                aria-hidden="true"
                className="icon ri-arrow-right-s-line"
                onClick={[Function]}
              />
            </div>
            <div
              className="active section"
              onClick={[Function]}
            >
              My first blog
            </div>
          </li>
        </ol>
      </nav>
    </div>
  </div>,
]
`;

exports[`Breadcrumbs renders breadcrumbs with contentTypesAsBreadcrumbSection 1`] = `
Array [
  <div
    id="page-header"
  />,
  <div
    className="ui vertical attached segment breadcrumbs"
  >
    <div
      className="ui container"
    >
      <nav
        aria-label="Breadcrumbs"
        className="ui tiny breadcrumb"
      >
        <ol
          aria-label="Breadcrumbs navigation"
        >
          <li
            className="breadcrumbs-home-item"
          >
            <a
              className="section"
              href="/"
              onClick={[Function]}
              title="Home"
            >
              <i
                aria-hidden="true"
                className="icon ri-home-2-line breadcrumbs-home-icon"
                onClick={[Function]}
              />
            </a>
          </li>
          <li>
            <div
              className="divider"
            >
              <i
                aria-hidden="true"
                className="icon ri-arrow-right-s-line"
                onClick={[Function]}
              />
            </div>
            <a
              className="section"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <div
              className="divider"
            >
              <i
                aria-hidden="true"
                className="icon ri-arrow-right-s-line"
                onClick={[Function]}
              />
            </div>
            <div
              className="section"
              onClick={[Function]}
            >
              News
            </div>
          </li>
          <li>
            <div
              className="divider"
            >
              <i
                aria-hidden="true"
                className="icon ri-arrow-right-s-line"
                onClick={[Function]}
              />
            </div>
            <div
              className="active section"
              onClick={[Function]}
            >
              Latest Update
            </div>
          </li>
        </ol>
      </nav>
    </div>
  </div>,
]
`;
